data/kernels/basic.cl

```diff
@@ -313,15 +313,22 @@ static float _calc_refavg(
         global const unsigned char (*const xtrans)[6],
         const unsigned int filters,
         int row,
-        int col)
+        int col,
+        int maxrow,
+        int maxcol)
 {
   float mean[3] = { 0.0f, 0.0f, 0.0f };
   float sum[3] =  { 0.0f, 0.0f, 0.0f };
   float cnt[3]  = { 0.0f, 0.0f, 0.0f };
```

Comment:
For a separate PR: would it make sense to declare these variables as dt_aligned_pixel_t?

Reply:
From my current understanding how CL compilers work i would say no. One of the tasks they do is to translate as much variables as possible to internal registers as memory is very restricted performance-wise. That's roughly what i got from reading OpenCL "best practice suggestions". But - never tested that personally.

Reply:
BTW: Just one thing i wanted to mention; have been away for 10days and just saw an "endless stream" of commits related to SSE2 code on my phone. Now back - just one word: CONGRATS!

Reply:
Thanks!   One more PR coming this weekend with a rewrite of the pixel interpolation code (sneak preview: 20+% speedup and removal of the majority of the SSE code).  Need to triple-check that I haven't broken anything.

Didn't notice that I was commenting on a .cl file.  I suspect that the CL compiler wouldn't support alignment directives anyway.